adapter/internal/oasparser/envoyconf/routes_configs.go

```diff
@@ -75,27 +75,35 @@ func generateRouteMatch(routeRegex string) *routev3.RouteMatch {
 	return match
 }
 
-func generateRouteAction(apiType string, prodRouteConfig, sandRouteConfig *model.EndpointConfig) (action *routev3.Route_Route) {
+func generateRouteAction(apiType string, prodRouteConfig, sandRouteConfig *model.EndpointConfig, endpointType string) (action *routev3.Route_Route) {
 
 	config, _ := config.ReadConfigs()
 
 	action = &routev3.Route_Route{
 		Route: &routev3.RouteAction{
-			HostRewriteSpecifier: &routev3.RouteAction_AutoHostRewrite{
-				AutoHostRewrite: &wrapperspb.BoolValue{
-					Value: true,
-				},
-			},
 			UpgradeConfigs:    getUpgradeConfig(apiType),
 			MaxStreamDuration: getMaxStreamDuration(apiType),
 			Timeout:           durationpb.New(time.Duration(config.Envoy.Upstream.Timeouts.RouteTimeoutInSeconds) * time.Second),
 			IdleTimeout:       durationpb.New(time.Duration(config.Envoy.Upstream.Timeouts.RouteIdleTimeoutInSeconds) * time.Second),
-			ClusterSpecifier: &routev3.RouteAction_ClusterHeader{
-				ClusterHeader: clusterHeaderName,
-			},
 		},
 	}
 
+	if endpointType == constants.AwsLambda {
+		action.Route.ClusterSpecifier = &routev3.RouteAction_Cluster{
+			Cluster: "wso2_lambda_egress_gateway",
```

Comment:
Can we use the constant you introduced `awslambdaClusterName` in here ?

Reply:
Fixed [https://github.com/wso2/product-microgateway/pull/3278/commits/4c71bdf884a8565b99b86e165aef47dad6489c0c](https://github.com/wso2/product-microgateway/pull/3278/commits/4c71bdf884a8565b99b86e165aef47dad6489c0c)